src/combinatorics/simple_graph/basic.lean

```diff
@@ -576,4 +592,175 @@ end
 
 end complement
 
+section maps
+
+/--
+A graph homomorphism is a map on vertex sets that respects adjacency relations.
+
+The notation `G →g G'` represents the type of graph homomorphisms.
+-/
+abbreviation hom := rel_hom G.adj G'.adj
+
+/--
+A graph embedding is an embedding `f` such that for vertices `v w : V`,
+`G.adj f(v) f(w) ↔ G.adj v w `. Its image is an induced subgraph of G'.
+
+The notation `G ↪g G'` represents the type of graph embeddings.
+-/
+abbreviation embedding := rel_embedding G.adj G'.adj
+
+/--
+A graph isomorphism is an bijective map on vertex sets that respects adjacency relations.
+
+The notation `G ≃g G'` represents the type of graph isomorphisms.
+-/
+abbreviation iso := rel_iso G.adj G'.adj
+
+infix ` →g ` : 50 := hom
+infix ` ↪g ` : 50 := embedding
+infix ` ≃g ` : 50 := iso
+
+namespace hom
+variables {G G'} (f : G →g G')
+
+/-- The identity homomorphism from a graph to itself. -/
+abbreviation id : G →g G := rel_hom.id _
+
+lemma map_adj {v w : V} (h : G.adj v w) : G'.adj (f v) (f w) := f.map_rel' h
+
+/-- The map between edge sets induced by a homomorphism.
+The underlying map on edges is given by `sym2.map`. -/
+@[simps] def map_edge_set (e : G.edge_set) : G'.edge_set :=
```

Comment:
This lemma uses the sets coerced to types, but I think there's also value in having results about `e : sym2 V` with the predicate that it's in `G.edge_set`. For instance, perhaps there should be a lemma saying
`example (e : sym2 V) (he : e ∈ G.edge_set) : e.map f ∈ G'.edge_set`
and you can use that lemma in this definition. And similarly for the other constructions here.

Reply:
Thanks, I think I've split off all the corresponding lemmas now.